### PR TITLE
Improve handling of WISPR maps

### PR DESCRIPTION
**Pull request by svank**
At @Cadair 's suggestion, this PR adds nice default colorbar vmin/vmax values for PSP/WISPR maps. Without this, the bright stars cause the automatic bounds to leave the faint coronal structures mostly invisible.

The range of brightness values in a WISPR image varies significantly through each encounter, so no one set of vmin/vmax defaults will be perfect, but I've found these values to be generally good defaults in my own work.

Also offers nicer names for the two detectors---WISPR is two imagers, an inner and an outer FOV. The FITS `detector` value is `1` or `2` to indicate the source imager, but providing labels of `Inner` and `Outer` for e.g. plot titles is nicer and more in line with how the community seems to refer to the two.

Before:
`wispr_map.peek()`
image

After:
`wispr_map.peek()`
image

### CODE REVIEW

**Comment by Cadair** [comment]
Thanks a lot @svank I will review it properly next week.

Makes me wonder exactly how much of a pain changing our default units would be (a lot).

**Comment by ayshih** [comment]
I'm hesitant on hard-coding vmin/vmax, and a user could always use `clip_interval`.  How well does `AsinhStretch` work instead of hard-coding vmin/vmax?

**Review by dstansby — CHANGES_REQUESTED** [review]
Looks good to me 👍 

Before this is merged, could you add a changelog entry?

**Comment by svank** [comment]
@dstansby Done, thanks!

**Review by dstansby — APPROVED** [review]

**Comment by ayshih** [comment]
I'll reiterate:

> I'm hesitant on hard-coding vmin/vmax, and a user could always use `clip_interval`.

We've (historically) followed the philosophy that showing all of the map data was preferable to showing a "pretty" map with defaults.  It's difficult to go outright "wrong" by showing the user all of the data, but you can confuse users about what's in the data by preemptively clipping out some of the data.  Plus, this avoided arguments about at what exact level to clip for, say, AIA maps.

Admittedly, this interest of data evaluation is partially satisfied (now) by `quicklook()` with its histogram equalization, so maybe it's okay to allow for clipping in the map sources.  However, I feel that this PR should follow the current philosophy, with a potential broad change to this philosophy as a subsequent PR (i.e., that changes the WISPR subclass in tandem with many other subclasses).

**Comment by ayshih** [comment]
(We have followed this philosophy since #1432 and `sunpy` 0.6)

File: sunpy/map/sources/psp.py
Comment (wtbarnes) [inline sunpy/map/sources/psp.py:75]: If the only allowed cases are 1 or 2, then this should be strictly enforced, i.e. this should fail if detector does not equal 1 or 2. 

```suggestion
        if detector == 1:
            return "Inner"
        elif detector == 2:
            return "Outer"
        else:
            raise ValueError(f'Unrecognized detector type {detector}')
```

**Comment by wtbarnes** [comment]
@ayshih  Do you want to block this until the debate about setting hard limits on the normalization has been settled? If so, I'd suggest explicitly requesting changes on this PR.

File: sunpy/map/sources/psp.py
Comment (wtbarnes) [inline sunpy/map/sources/psp.py:49]: What case is this try-except catching? Are there files that don't have the detector or level set?

**Review by ayshih — CHANGES_REQUESTED** [review]
> @ayshih Do you want to block this until the debate about setting hard limits on the normalization has been settled? If so, I'd suggest explicitly requesting changes on this PR.

Well, if you're going to make me...

\<in booming voice\>

_No map-source subclass shall specify a default value of `vmin`/`vmax` for plotting that can alter the apparent value for any pixel (i.e., via clipping) without providing due notice to the user in the docstring of said map-source subclass.  That notice shall include a justification for the value chosen for `vmin`/`vmax`._

\<deep breath\>

_If the chosen value is time-dependent (e.g., because it depends on instrument calibration), or has the potential to need to be updated in the future, the notice shall cite a person, guide, or publication with the authority (explicit or implied) to represent the standard practice (official or de facto) in the community for analyzing said map source._

File: sunpy/map/sources/psp.py
Comment (svank) [reply sunpy/map/sources/psp.py:49]: Yeah, it's a failsafe for if the detector or level isn't in the collection of presets. For this and your other comment, the official data products should only ever have `1` or `2` for the `detector` entry in the header (I'm not a SOC member that can *guarantee* this, but I don't think they'll unveil a secret third detector any time soon). The `level` entry can also be `L1` or `L2b`---this PR was mostly just copying over snippets from my own WISPR tools, and since I don't work with those levels I didn't have presets ready, but they could be prepared.

Outside the official data products, others on the team produce `LW` and `LT` files using other processing techniques, and they put `LW` in the `level` field, for instance. One could also imagine someone producing a composite from the two imagers and putting some other value in the `detector` field to indicate this (or deleting the field altogether).

So I think either `WISPRMap` should be able to gracefully handle unexpected values in these fields, or `is_datasource_for` should only return `True` if these fields contain expected values. Not sure which is preferable.

**Comment by svank** [comment]
I'm waiting for you maintainers to settle this debate. I think the coronal signal (versus the background stars) is very likely the signal of interest for almost all users, so using the full colorbar for that by default seems sensible, and the fixed vmin/vmax seems to me the best way to do that, but `AsinhStretch` was at least OK and didn't require those fixed values and I see the wisdom in not hiding new philosophies in this little PR.

**Comment by ayshih** [comment]
There isn't actually any debate among the maintainers; there's broad agreement with my above decree.

As it pertains to this PR:

- `vmin`
  - Setting `vmin` to zero should be documented in the class docstring with a brief justification.
  - No reference is required as zero is a special value that is unlikely to require future updating.
- `vmax`
  - Setting `vmax` to a non-zero value is allowed only if the choice is backed by authority or reference.
  - The value, the justification, and the authority/reference should be documented in the class docstring.

In the absence of adequate authority/reference for the setting of `vmax`, I suggest that this PR go forward with `vmin=0` and `AsinhStretch`.

**Comment by svank** [comment]
OK, here's a version with `AsinhStretch`. I spot-checked for L2 and L3 data from both detectors, and I think the plots look reasonable. The apparent background level can vary through an encounter, sometimes very rapidly, as `vmax` floats but I think that's just the way it is.

After @wtbarnes 's comments, I've clarified my intent in `detector()` about handling unexpected values. I also applied similar logic to `processing_level()`, ensuring that the `int` conversion fails safe for non-`int` values (which I know people use for files outside the official data products). At the risk of broadening this PR too much, I introduced one change in behavior, which I think counts as a bug fix. The official data files include `L2b` files, which are the estimated backgrounds subtracted from `L2` files to produce `L3` files. In the FITS headers, the `LEVEL` field is `L2`, `L2b`, or `L3`. Before, `processing_level()` would chop off the leading `L` by returning `int(lvl[1])`, providing a level of `2` for both `L2` and `L2b` files. Now it parses `lvl[1:]` instead, so for `L2` files it will return `2` as an int, the same as before, and it returns `2b` as a string for `L2b` files. If `processing_level()` is meant to be just convenient access to the `LEVEL` field, this ensures we're not accidentally providing the same parsed value for files with different values in the actual header.

File: sunpy/map/sources/psp.py
Comment (ayshih) [inline sunpy/map/sources/psp.py:31]: Why are you choosing to set `clip=True`?  That is more aggressive than merely setting `vmin`.

File: sunpy/map/sources/psp.py
Comment (ayshih) [inline sunpy/map/sources/psp.py:40]:
```suggestion
        # Chop off the leading 'L' if present
        lvl = lvl[1:] if lvl[0] == 'L' else lvl
```
Since you are allowing for custom levels, it seems odd not to check for whether the character being popped off is 'L' or not.

File: sunpy/map/sources/psp.py
Comment (ayshih) [inline sunpy/map/sources/psp.py:46]:
```suggestion
        try:
            lvl = int(lvl)
        except ValueError:
            # The int conversion will fail for L2b files, and we should fail
            # safe if the user chooses to customize this with other values.
            pass
        return lvl
```
I'm probably just being crotchety, but I don't like seeing the `return` inside a `try...except` here because the `return` will never raise an exception.

**Review by ayshih — COMMENTED** [review]
Add a note to the docstring before the `References` section a la:
```
Notes
-----
By default, plotting of this map will set the lower bound to zero
(i.e., clip out negative values for pixels).  You can change this bound
by modifying ``.plot_settings['norm'].vmin``.

**Review by wtbarnes — APPROVED** [review]
Thanks @svank! This looks good to me.

File: sunpy/map/sources/psp.py
Comment (svank) [reply sunpy/map/sources/psp.py:31]: Good catch, I think this was just left from the initial, more opinionated plot settings.

**Comment by svank** [comment]
@ayshih I've made your requested changes, thanks!

**Review by ayshih — APPROVED** [review]
Looks good to me, thanks!

**Comment by ayshih** [comment]
Also, thanks for your patience!

**Pull request merged**
